libraries/Microsoft.Bot.Builder.Azure/CosmosDbStorage.cs

```diff
@@ -37,41 +37,31 @@ public class CosmosDbStorage : IStorage
         /// Creates a new <see cref="CosmosDbStorage"/> object,
         /// using the provided CosmosDB credentials, database ID, and collection ID.
         /// </summary>
-        /// <param name="serviceEndpoint">The URI of the service endpoint for the Azure Cosmos DB service.</param>
-        /// <param name="authKey">The AuthKey used by the client from the Azure Cosmos DB service.</param>
-        /// <param name="databaseId">The database ID.</param>
-        /// <param name="collectionId">The collection ID.</param>
-        /// <param name="connectionPolicyConfigurator">A connection policy delegate.</param>
-        /// <remarks>
-        /// You can use the <paramref name="connectionPolicyConfigurator"/> delegate to 
-        /// further customize the connection to CosmosDB, 
-        /// such as setting connection mode, retry options, timeouts, and so on.
-        /// See https://docs.microsoft.com/en-us/dotnet/api/microsoft.azure.documents.client.connectionpolicy?view=azure-dotnet
-        /// for more information.</remarks>
-        public CosmosDbStorage(Uri serviceEndpoint, string authKey, string databaseId, string collectionId, Action<ConnectionPolicy> connectionPolicyConfigurator = null)
+        /// <param name="cosmosDbStorageOptions">Cosmos DB storage configuration options.</param>
+        public CosmosDbStorage(CosmosDbStorageOptions cosmosDbStorageOptions)
```

Comment:
Consider adding the basic null check:
if (cosmosDbStorageOptions == null) 
    throw new ArgumentNullException(...)